From efa26eaf4fba939c3e43b659e5c77a57b22b270a Mon Sep 17 00:00:00 2001
From: Georg Koppen <gk@torproject.org>
Date: Fri, 2 Oct 2020 08:48:49 +0000
Subject: [PATCH] Update glean for tor-browser-build#40085


diff --git a/buildSrc/src/main/java/Dependencies.kt b/buildSrc/src/main/java/Dependencies.kt
index f4063f7ad..245d14447 100644
--- a/buildSrc/src/main/java/Dependencies.kt
+++ b/buildSrc/src/main/java/Dependencies.kt
@@ -29,7 +29,7 @@ object Versions {
 
     const val mozilla_appservices = "61.0.13"
 
-    const val mozilla_glean = "32.1.1"
+    const val mozilla_glean = "32.4.1"
 
     const val material = "1.1.0"
     const val nearby = "17.0.0"
-- 
2.28.0

