From a77b847b6451c598176137b944edf27e6e7202a2 Mon Sep 17 00:00:00 2001
From: Georg Koppen <gk@torproject.org>
Date: Sun, 19 Apr 2020 17:59:16 +0000
Subject: [PATCH 1/5] git patch


diff --git a/components/support/base/build.gradle b/components/support/base/build.gradle
index aa4c1b005..4318f1334 100644
--- a/components/support/base/build.gradle
+++ b/components/support/base/build.gradle
@@ -37,7 +37,7 @@ android {
             minifyEnabled false
             proguardFiles getDefaultProguardFile('proguard-android.txt'), 'proguard-rules.pro'
 
-            buildConfigField("String", "GIT_HASH", "\"" + getGitHash() + "\"")
+            buildConfigField("String", "GIT_HASH", "\"release build\"")
         }
     }
 
-- 
2.27.0.rc0

