From 6a88b59457dd38f0e8d69f0c2988bb2952075536 Mon Sep 17 00:00:00 2001
From: Georg Koppen <gk@torproject.org>
Date: Wed, 20 May 2020 14:18:12 +0000
Subject: [PATCH] mavenLocal() fixes


diff --git a/build.gradle b/build.gradle
index 9f298887a..278245d30 100644
--- a/build.gradle
+++ b/build.gradle
@@ -6,6 +6,7 @@ import static org.gradle.api.tasks.testing.TestResult.ResultType
 
 buildscript {
     repositories {
+        mavenLocal()
         if (project.hasProperty("googleRepo")) {
             maven {
                 name "Google"
@@ -48,6 +49,7 @@ plugins {
 
 allprojects {
     repositories {
+        mavenLocal()
         if (project.hasProperty("googleRepo")) {
             maven {
                 name "Google"
diff --git a/buildSrc/build.gradle b/buildSrc/build.gradle
index 9f05b8434..f97029ccf 100644
--- a/buildSrc/build.gradle
+++ b/buildSrc/build.gradle
@@ -7,6 +7,7 @@ plugins {
 }
 
 repositories {
+    mavenLocal()
     jcenter()
 }
 
diff --git a/buildSrc/settings.gradle b/buildSrc/settings.gradle
index c31ad3840..fb6526db3 100644
--- a/buildSrc/settings.gradle
+++ b/buildSrc/settings.gradle
@@ -1,10 +1,15 @@
 /* This Source Code Form is subject to the terms of the Mozilla Public
  * License, v. 2.0. If a copy of the MPL was not distributed with this
  * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+pluginManagement {
+  repositories {
+    mavenLocal()
+  }
+}
 
 buildCache {
     local {
         directory = new File(rootDir, '../.build-cache')
         removeUnusedEntriesAfterDays = 30
     }
-}
\ No newline at end of file
+}
diff --git a/settings.gradle b/settings.gradle
index a6e8dfa49..049fa29d6 100644
--- a/settings.gradle
+++ b/settings.gradle
@@ -4,8 +4,15 @@
 
 import org.yaml.snakeyaml.Yaml
 
+pluginManagement {
+  repositories {
+    mavenLocal()
+  }
+}
+
 buildscript {
     repositories {
+        mavenLocal()
         jcenter()
     }
 
-- 
2.28.0

