diff -Naur nsis-3.08-orig/SCons/Config/gnu nsis-3.08-src/SCons/Config/gnu
--- nsis-3.08-orig/SCons/Config/gnu	2023-03-23 09:22:46.315471779 +0100
+++ nsis-3.08-src/SCons/Config/gnu	2023-03-23 09:24:05.260933879 +0100
@@ -103,6 +103,10 @@
 stub_env.Append(LINKFLAGS = ['$ALIGN_FLAG'])        # 512 bytes align
 stub_env.Append(LINKFLAGS = ['$MAP_FLAG'])          # generate map file

+# https://sourceforge.net/p/nsis/bugs/1283/?limit=25#e303,
+# https://sourceforge.net/p/nsis/bugs/1283/?limit=25#e303/e90f
+stub_env.Append(LINKFLAGS = ['-Wl,--disable-reloc-section'])
+
 conf = FlagsConfigure(stub_env)
 conf.CheckCompileFlag('-fno-tree-loop-distribute-patterns')  # GCC 10: Don't generate msvcrt!memmove calls (bug #1248)
 conf.Finish()
