diff --git a/CMakeLists.txt b/CMakeLists.txt
index 79d4302..0ddf545 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -14,6 +14,10 @@ IF(BUILD_STATIC)
 ENDIF(BUILD_STATIC)
 
 include_directories (${PROJECT_SOURCE_DIR}/includes)
+include_directories(${ZLIB_INCLUDE_DIR})
+link_directories(${ZLIB_LIBRARIES})
+include_directories(${BZIP2_INCLUDE_DIR})
+link_directories(${BZIP2_LIBRARIES})
 
 add_subdirectory (common)
 add_subdirectory (dmg)
diff --git a/dmg/CMakeLists.txt b/dmg/CMakeLists.txt
index f6df8fe..1e85b9f 100644
--- a/dmg/CMakeLists.txt
+++ b/dmg/CMakeLists.txt
@@ -1,19 +1,3 @@
-# INCLUDE(FindOpenSSL)
-INCLUDE(FindZLIB)
-INCLUDE(FindBZip2)
-
-IF(NOT ZLIB_FOUND)
-	message(FATAL_ERROR "zlib is required for dmg!")
-ENDIF(NOT ZLIB_FOUND)
-IF(NOT BZIP2_FOUND)
-	message(FATAL_ERROR "bzip2 is required for dmg!")
-ENDIF(NOT BZIP2_FOUND)
-
-include_directories(${ZLIB_INCLUDE_DIR})
-link_directories(${ZLIB_LIBRARIES})
-include_directories(${BZIP2_INCLUDE_DIR})
-link_directories(${BZIP2_LIBRARIES})
-
 link_directories(${PROJECT_BINARY_DIR}/common ${PROJECT_BINARY_DIR}/hfs)
 
 add_library(dmg adc.c attribution.c checksum.c dmgfile.c dmglib.c filevault.c io.c partition.c resources.c udif.c ../includes/dmg/adc.h ../includes/dmg/attribution.h ../includes/dmg/dmg.h ../includes/dmg/dmgfile.h ../includes/dmg/dmglib.h ../includes/dmg/filevault.h)
diff --git a/hfs/CMakeLists.txt b/hfs/CMakeLists.txt
index 8bb2a52..6807ca1 100644
--- a/hfs/CMakeLists.txt
+++ b/hfs/CMakeLists.txt
@@ -1,12 +1,3 @@
-INCLUDE(FindZLIB)
-
-IF(NOT ZLIB_FOUND)
-	message(FATAL_ERROR "zlib is required for hfs!")
-ENDIF(NOT ZLIB_FOUND)
-
-include_directories(${ZLIB_INCLUDE_DIR})
-link_directories(${ZLIB_LIBRARIES})
-
 link_directories (${PROJECT_BINARY_DIR}/common)
 add_library(hfs btree.c catalog.c extents.c xattr.c fastunicodecompare.c flatfile.c hfslib.c rawfile.c utility.c volume.c hfscompress.c ../includes/hfs/hfslib.h ../includes/hfs/hfsplus.h ../includes/hfs/hfscompress.h)
 target_link_libraries(hfs common z)
