diff --git a/CMakeLists.txt b/CMakeLists.txt
index 79d4302..0ddf545 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -14,6 +14,10 @@ IF(BUILD_STATIC)
 ENDIF(BUILD_STATIC)
 
 include_directories (${PROJECT_SOURCE_DIR}/includes)
+include_directories(${ZLIB_INCLUDE_DIR})
+link_directories(${ZLIB_LIBRARIES})
+include_directories(${BZIP2_INCLUDE_DIR})
+link_directories(${BZIP2_LIBRARIES})
 
 add_subdirectory (common)
 add_subdirectory (dmg)
diff --git a/dmg/CMakeLists.txt b/dmg/CMakeLists.txt
index 6250a6d..5b438b7 100644
--- a/dmg/CMakeLists.txt
+++ b/dmg/CMakeLists.txt
@@ -1,18 +1,4 @@
 INCLUDE(FindOpenSSL)
-INCLUDE(FindZLIB)
-INCLUDE(FindBZip2)
-
-IF(NOT ZLIB_FOUND)
-	message(FATAL_ERROR "zlib is required for dmg!")
-ENDIF(NOT ZLIB_FOUND)
-IF(NOT BZIP2_FOUND)
-	message(FATAL_ERROR "bzip2 is required for dmg!")
-ENDIF(NOT BZIP2_FOUND)
-
-include_directories(${ZLIB_INCLUDE_DIR})
-link_directories(${ZLIB_LIBRARIES})
-include_directories(${BZIP2_INCLUDE_DIR})
-link_directories(${BZIP2_LIBRARIES})
 
 link_directories(${PROJECT_BINARY_DIR}/common ${PROJECT_BINARY_DIR}/hfs)
 
@@ -30,7 +16,7 @@ ENDIF(OPENSSL_FOUND)
 target_link_libraries(dmg common hfs z bz2)
 
 add_executable(dmg-bin dmg.c)
-target_link_libraries (dmg-bin dmg)
+target_link_libraries (dmg-bin dmg pthread)
 
 set_target_properties(dmg-bin PROPERTIES OUTPUT_NAME "dmg")
 
diff --git a/hdutil/CMakeLists.txt b/hdutil/CMakeLists.txt
index f0d8ac9..f5f0f62 100644
--- a/hdutil/CMakeLists.txt
+++ b/hdutil/CMakeLists.txt
@@ -2,7 +2,7 @@ link_directories(${PROJECT_BINARY_DIR}/common ${PROJECT_BINARY_DIR}/hfs ${PROJEC
 
 add_executable(hdutil hdutil.c)
 
-target_link_libraries (hdutil dmg hfs common)
+target_link_libraries (hdutil dmg hfs common pthread)
 
 install(TARGETS hdutil DESTINATION bin)
 
